### PR TITLE
Added rate limit to the logins

### PR DESCRIPTION
**Pull request by nakel-ola**

### CODE REVIEW

File: app/api/auth/login/gallery/route.ts [hunk 1]
```diff
@@ -1,5 +1,9 @@
-import { ConflictError } from "@/custom/errors/dictionary/errorDictionary";
+import {
+  ConflictError,
+  TooManyRequestError,
```
Comment (Moses-py) [inline app/api/auth/login/gallery/route.ts:3]: Can we change this from "TooManyRequestError" to "RateLimitExceededErrror" in all instances of it's usage? Just so it's more in line with conventional naming patterns

File: app/api/auth/login/gallery/route.ts [hunk 2]
```diff
@@ -10,6 +14,11 @@ import { NextResponse, NextResponse as res } from "next/server";
 
 export async function POST(request: Request) {
   try {
+    const remainingRequests = await limiter.removeTokens(1);
+
+    if (remainingRequests < 0)
+      throw new TooManyRequestError("Too many Requests");
```
Comment (Moses-py) [inline app/api/auth/login/gallery/route.ts:20]: Please use "RateLimitExceededErrror" instead.

File: app/api/auth/login/individual/route.ts [hunk 3]
```diff
@@ -10,6 +14,11 @@ import { NextResponse, NextResponse as res } from "next/server";
 
 export async function POST(request: Request) {
   try {
+    const remainingRequests = await limiter.removeTokens(1);
+
+    if (remainingRequests < 0)
+      throw new TooManyRequestError("Too many Requests");
```
Comment (Moses-py) [inline app/api/auth/login/individual/route.ts:20]: Please use "RateLimitExceededErrror" instead.

**Review by Moses-py — CHANGES_REQUESTED** [review]

File: app/api/auth/login/gallery/route.ts (hunk 1 above)
Comment (nakel-ola) [reply app/api/auth/login/gallery/route.ts:3]: Okay

**Review by Moses-py — APPROVED** [review]

**Pull request merged**